runner/main.go

```diff
@@ -21,7 +20,6 @@ type (
 		panicAsFail      bool
 		logger           *log.Logger
 		failed           bool
-		failedlock       *sync.Mutex
```

Comment:
What changed for this to be now not needed? Or it wasn't needed all along?

Reply:
Since I added this: 
https://github.com/undefinedlabs/scope-go-agent/blob/fdb5a3e72021df0e4ca4248d889ece945152a8cb/runner/main.go#L100-L103

All outer `testing.T` are sequential, because no `Parallel` is set in those outer tests.

Reply:
What you mean with "outer `testing.T`? (Sorry, I need context :-S)

Reply:
Ok, the runner works by replacing the original test method with a wrapper, that wrapper creates a subtest and run the original test there, when the subtest finishes we grab the results and checks if fails / pass / panic. then in this outer test or wrapper we decide if we run again the original test as another subtest, to detect flakiness for example...

Because this outter or wrapper test are not marked as `t.Parallel` the execution will be sequential, and because is sequential, this line doesn't need locks:

https://github.com/undefinedlabs/scope-go-agent/blob/runner-lock/runner/main.go#L159-L163

because the parent of the wrapper will be modified by a single running test

Reply:
We can keep the lock but i think is no needed

Reply:
Can we detect if it is needed using the `go test -race` in the go-agent?

Reply:
I've just rebase from master, now the test will run with `go test -v -race ./...`